src/workerd/io/worker-fs.h

```diff
@@ -164,6 +164,15 @@ struct Stat final {
 
 class SymbolicLink;
 
+enum class FsError {
+  // Path segment is not a directory
+  NOT_DIRECTORY,
+  // Directory is not empty
+  NOT_EMPTY,
+  // Node is read-only
+  READ_ONLY,
+};
```

Comment:
This will be expanded further.. I'm going to be moving the worker-fs impl away from throwing js errors directly to returning error codes since the node.js error codes are going to differ from the web fs api errors